tests/base.py

```diff
@@ -133,15 +133,14 @@ def _add_gids_per_port(self, ctx, dev, port):
         # Don't add ports which are not active
         if ctx.query_port(port).state != e.IBV_PORT_ACTIVE:
             return
-        idx = 0
-        while True:
+
+        for idx in range(ctx.query_port(port).gid_tbl_len):
             gid = ctx.query_gid(port, idx)
             if gid.gid[-19:] == self.ZERO_GID:
-                # No point iterating on
-                break
```

Comment:
It is correct now, but just write it more simply as

if gid.gid[-19:] != self.ZERO_GID:
   self.args.append([dev, port, idx])

and why does this append an array not a tuple? Strange.

Reply:
> It is correct now, but just write it more simply as
> 
> if gid.gid[-19:] != self.ZERO_GID:
> self.args.append([dev, port, idx])
> 
OK, I'll update the patch.

> and why does this append an array not a tuple? Strange.

Yes, that is strange, I agree that this need to be a tuple to avoid confusion, but probably in a different patch.